FastPriorityQueue.js

```diff
@@ -287,13 +287,22 @@ FastPriorityQueue.prototype.forEach = function(callback) {
 FastPriorityQueue.prototype.kSmallest = function(k) {
   if (this.size == 0) return [];
   k = Math.min(this.size, k);
-  var fpq = new FastPriorityQueue(this.compare);
-  const newSize = Math.min((k > 0 ? Math.pow(2, k - 1) : 0) + 1, this.size);
+  const newSize = Math.min(this.size, (1 << (k - 1)) + 1);
+  if (newSize < 2) {
+    if (newSize < 1)  {
+      return [];
+    } else if (k > 0) {
+      return [this.peek()];
+    }
+  }
+  if (k < 1) return [];
```

Comment:
If you guard at the start for this, as I suggest, then this line is unneeded.

Reply:
`k < 1` check removed.